uk.co.compucorp.civicrm.hrcore/CRM/HRCore/CMSData/CMSPathsFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+class CRM_HRCore_CMSData_CMSPathsFactory {
+
+  /**
+   * Instantiates a Paths class based on the given CMS name
+   *
+   * @param string $cmsName
+   * @param array $contactData
+   *
+   * @return object
+   *
+   * @throws Exception if the CMS is not recognized
+   */
+  public static function create($cmsName, $contactData) {
+    $allowedCms = ['Drupal', 'WordPress', 'Joomla'];
```

Comment:
Since you kept only the Drupal class, it's better to remove the references to other CMSes in here